server/tserver/src/main/java/org/apache/accumulo/tserver/replication/AccumuloReplicaSystem.java

```diff
@@ -416,6 +444,12 @@ protected Status replicateLogs(ClientContext peerContext, final HostAndPort peer
     Status lastStatus = status, currentStatus = status;
     final AtomicReference<Exception> exceptionRef = new AtomicReference<>();
     while (true) {
+      // Check if the thread has been interrupted prior to replicating data
+      if (Thread.interrupted()) {
```

Comment:
I am seeing the interrupt being caught by this thread and properly handling the timeout.  I can add a boolean regardless, though, if that is the consensus.